clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -200,32 +201,38 @@ public void run() {
      */
     void run(long now) {
         if (transactionManager != null) {
-            if (transactionManager.shouldResetProducerStateAfterResolvingSequences())
```

Comment:
@hachikuji, if you have a chance, have a look and see if this is OK.

Reply:
Looks ok to me.